h5py/tests/test_completions.py

```diff
@@ -1,40 +1,37 @@
-from .common import TestCase
+import h5py
+
+from .common import TestCase, make_name
 
 
 class TestCompletions(TestCase):
 
-    def test_group_completions(self):
+    def test_root_group_completions(self):
         # Test completions on top-level file.
-        g = self.f.create_group('g')
-        self.f.create_group('h')
-        self.f.create_dataset('data', [1, 2, 3])
-        self.assertEqual(
-            self.f._ipython_key_completions_(),
-            ['data', 'g', 'h'],
-        )
-
-        self.f.create_group('data2', [1, 2, 3])
-        self.assertEqual(
-            self.f._ipython_key_completions_(),
-            ['data', 'data2', 'g', 'h'],
-        )
-
-        # Test on subgroup.
-        g.create_dataset('g_data1', [1, 2, 3])
+        with h5py.File(self.mktemp(), 'w') as f:
+            f.create_group('h')
+            f.create_group('g')
+            f.create_group('data2')
+            f.create_dataset('data', [1, 2, 3])
+
+            # Test that order is alphabetical, and that there is no
+            # internal ordering between groups and datasets.
+            self.assertEqual(
+                f._ipython_key_completions_(),
+                ['data', 'data2', 'g', 'h'],
+            )
+
+    def test_subgroup_completions(self):
+        g = self.f.create_group(make_name())
         g.create_dataset('g_data2', [4, 5, 6])
-        self.assertEqual(
-            g._ipython_key_completions_(),
-            ['g_data1', 'g_data2'],
-        )
-
```

Comment:
I think the testing here was about caching and that creating a 3rd group would not still return `['g_data1', 'g_data2']` and indeed have             `'g_data3'`